specs/packagemanager/PackageManagement.md

```diff
@@ -635,6 +644,66 @@ void CheckAndFixPackageVolume(string packageStorePath)
 }
 ```
 
+## 4.9. IsPackageProvisioned()
+
+Fabrikam app installing Contoso's Muffin and Waffle packages for all users if necessary, and with explicit user confirmation before the installation.
+
+```c#
+void Install()
+{
+    var packageSet = new PackageSet() {
+        Items = { new PackageSetItem() { PackageFamilyName = "contoso.muffin_1234567890abc",
+                                         MinVersion = ToVersion(1, 2, 3, 4),
+                                         PackageUri = new Uri("c:\\contoso\\muffin-1.2.3.4.msix") },
+                { new PackageSetItem() { PackageFamilyName = "contoso.waffle_1234567890abc",
+                                         MinVersion = ToVersion(2, 4, 6, 8),
+                                         PackageUri = new Uri("https://contoso.com/waffle-2.4.6.8.msix") } };
+
+    var packageDeploymentManager = PackageDeploymentManager.GetDefault();
+    if (!packageDeploymentManager.IsPackageSetProvisioned(packageSet))
+    {
+        bool ok = PromptUserForConfirmation();
```

Comment:
>PromptUserForConfirmation() [](http://example.com/codeflow?start=18&length=27)

how does user know which packages will install?

Reply:
You assume the user needs to know, or would understand.

I know at least one prominent and recent similar user experience where the user is asked to confirm some package work at the functional level without sharing the implementation details like package names. And to be honest, even I would have a hard time understanding what the specific packages mean, other than the half dozen packages collectively are needed to power a feature.

For non-interactive post-mortem reporting or analysis deployment generates logs of all its activity so telling a user "build car?" rather than "assemble engine, doors, spark plugs, timing belt, ..." can be a valid (and sometimes better) user experience.

Reply:
If user is already installing the app why do we need another prompt to install dependencies?